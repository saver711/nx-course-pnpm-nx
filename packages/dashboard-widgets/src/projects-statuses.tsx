import type { JSX } from 'react';
import { ProjectWithTasks } from '@tasker/database/model';
import { ProjectItem } from '@tasker/ui/project';

interface ProjectsStatusesProps {
  projects: ProjectWithTasks[];
}

export function ProjectsStatuses({
  projects,
}: ProjectsStatusesProps): JSX.Element {
  return (
    <>
      <h3 className="mb-4 text-lg font-semibold">Projects</h3>
      <p className="mb-4 text-sm text-gray-500">
        Active projects and their status
      </p>
      <div className="grid gap-4">
        {projects.map((p: ProjectWithTasks) => (
          <ProjectItem key={p.id} project={p} />
        ))}
      </div>
    </>
  );
}
