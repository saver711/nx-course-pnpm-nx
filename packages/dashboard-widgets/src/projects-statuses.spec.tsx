import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import { ProjectsStatuses } from './projects-statuses';

test('ProjectStatus: renders with empty', () => {
  const { baseElement } = render(<ProjectsStatuses projects={[]} />);

  expect(baseElement).toMatchSnapshot();
});

test('ProjectStatus: renders with projects', () => {
  const { baseElement } = render(
    <ProjectsStatuses
      projects={[
        {
          status: 'on_track',
          name: 'Engineering',
          id: 1,
          slug: 'ENG',
          Task: [],
        },
        { status: 'on_track', name: 'Design', id: 2, slug: 'UX', Task: [] },
      ]}
    />,
  );

  expect(baseElement).toMatchSnapshot();
});
