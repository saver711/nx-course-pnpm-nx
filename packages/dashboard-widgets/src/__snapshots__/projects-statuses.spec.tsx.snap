// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectStatus: renders with empty 1`] = `
<body>
  <div>
    <h3
      class="mb-4 text-lg font-semibold"
    >
      Projects
    </h3>
    <p
      class="mb-4 text-sm text-gray-500"
    >
      Active projects and their status
    </p>
    <div
      class="grid gap-4"
    />
  </div>
</body>
`;

exports[`ProjectStatus: renders with projects 1`] = `
<body>
  <div>
    <h3
      class="mb-4 text-lg font-semibold"
    >
      Projects
    </h3>
    <p
      class="mb-4 text-sm text-gray-500"
    >
      Active projects and their status
    </p>
    <div
      class="grid gap-4"
    />
  </div>
  <div>
    <h3
      class="mb-4 text-lg font-semibold"
    >
      Projects
    </h3>
    <p
      class="mb-4 text-sm text-gray-500"
    >
      Active projects and their status
    </p>
    <div
      class="grid gap-4"
    >
      <a
        class="flex items-center justify-between"
        href="/projects/ENG"
      >
        <div
          class="flex items-center gap-2"
        >
          <div
            class="h-2 w-2 rounded-full bg-green-300"
          />
          <span
            class="text-sm font-medium text-gray-600"
          >
            Engineering
          </span>
        </div>
        <span
          class="rounded-md border border-gray-200 border-gray-300 px-2 py-1 text-xs font-medium text-gray-600 dark:border-gray-800"
          data-testid="project-item-status"
        >
          On Track
        </span>
      </a>
      <a
        class="flex items-center justify-between"
        href="/projects/UX"
      >
        <div
          class="flex items-center gap-2"
        >
          <div
            class="h-2 w-2 rounded-full bg-green-300"
          />
          <span
            class="text-sm font-medium text-gray-600"
          >
            Design
          </span>
        </div>
        <span
          class="rounded-md border border-gray-200 border-gray-300 px-2 py-1 text-xs font-medium text-gray-600 dark:border-gray-800"
          data-testid="project-item-status"
        >
          On Track
        </span>
      </a>
    </div>
  </div>
</body>
`;
